actors/miner/tests/prove_commit.rs

```diff
@@ -340,3 +354,343 @@ fn prove_sectors_from_batch_pre_commit() {
         assert_eq!(&no_deal_pledge + &full_deal_pledge + &full_deal_pledge, st.initial_pledge);
     }
 }
+
+fn max_prove_commit_duration(policy: &Policy, seal_proof: RegisteredSealProof) -> ChainEpoch {
+    use RegisteredSealProof::*;
+    match seal_proof {
+        // PARAM_SPEC
+        StackedDRG2KiBV1 => EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        StackedDRG512MiBV1 => EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        StackedDRG8MiBV1 => EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        StackedDRG32GiBV1 => EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        StackedDRG64GiBV1 => EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+
+        // PARAM_SPEC
+        StackedDRG2KiBV1P1 => 30 * EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        StackedDRG512MiBV1P1 => 30 * EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        StackedDRG8MiBV1P1 => 30 * EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        StackedDRG32GiBV1P1 => 30 * EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        StackedDRG64GiBV1P1 => 30 * EPOCHS_IN_DAY + policy.pre_commit_challenge_delay,
+        Invalid(_) => panic!("max_prove_commit_duration not defined for invalid seal proofs."),
+    }
+}
+
+#[test]
+fn invalid_proof_rejected() {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.balance.replace(TokenAmount::from(BIG_BALANCE));
+
+    let precommit_epoch = PERIOD_OFFSET + 1;
+    rt.set_epoch(precommit_epoch);
+
+    h.construct_and_verify(&mut rt);
+    let deadline = h.deadline(&rt);
+
+    // Make a good commitment for the proof to target.
+    let sector_no = 100;
+    let params = h.make_pre_commit_params(
+        sector_no,
+        precommit_epoch - 1,
+        deadline.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period,
+        vec![1],
+    );
+    let precommit = h.pre_commit_sector(&mut rt, params, PreCommitConfig::empty(), true);
+
+    // Sector pre-commitment missing.
+    rt.set_epoch(precommit_epoch + rt.policy.pre_commit_challenge_delay + 1);
+    expect_abort(
+        ExitCode::USR_NOT_FOUND,
+        h.prove_commit_sector_and_confirm(
+            &mut rt,
+            &precommit,
+            h.make_prove_commit_params(sector_no + 1),
+            ProveCommitConfig::empty(),
+        ),
+    );
+    rt.reset();
+
+    // Too late.
+    rt.set_epoch(
+        precommit_epoch + max_prove_commit_duration(&rt.policy, precommit.info.seal_proof) + 1,
+    );
+    expect_abort(
+        ExitCode::USR_NOT_FOUND,
```

Comment:
this is a duplicate of the previous case instead of the too late case

Reply:
Fixed.